### PR TITLE
[macOS] Remove macOS 10.14 related code in the VE repo

### PR DESCRIPTION
**Pull request by V-Zabayrachny**
# Description
MacOS 10.14 is outdated. In this regard, we remove all the related code from VE repo and delete all build\release pipelines from https://dev.azure.com/mseng/AzureDevOps/ and https://github.visualstudio.com/virtual-environments

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/utils/utils.sh [hunk 1]
```diff
@@ -63,32 +63,16 @@ is_Catalina() {
...
-
-is_Less_Catalina() {
-    if is_Mojave; then
-        true
-    else
-        false
-    fi
-}
-
 is_Less_BigSur() {
```
Comment (miketimofeev) [inline images/macos/provision/utils/utils.sh:82]: Do we need this now? It is the same as IsCatalina

Comment (V-Zabayrachny) [reply images/macos/provision/utils/utils.sh:82]: That's a good idea, thank you.

File: images/macos/helpers/Common.Helpers.psm1 [hunk 2]
```diff
@@ -34,14 +34,11 @@ function Get-OSVersion {
     return [PSCustomObject]@{
         Version = $osVersion.Version
         Platform = $osVersion.Platform
-        IsMojave = $osVersionMajorMinor -eq "10.14"
         IsCatalina = $osVersionMajorMinor -eq "10.15"
         IsBigSur = $osVersionMajorMinor -eq "11.0"
         IsMonterey = $osMontereyVersion.Major -eq "12"
-        IsLessThanCatalina = [SemVer]$osVersion.Version -lt "10.15"
         IsLessThanBigSur = [SemVer]$osVersion.Version -lt "11.0"
```
Comment (miketimofeev) [inline images/macos/helpers/Common.Helpers.psm1:40]: This can be removed too

Comment (V-Zabayrachny) [reply images/macos/helpers/Common.Helpers.psm1]: That's a good idea, thank you.

**Review by MaksimZhukov — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Pull request merged**